File: 762. Prime Number of Set Bits in Binary Representation.cpp
 bool isPrime(int n){
        vector<bool>primes(n+1,1);
        primes[0] = 0 , primes[1] = 0;
        for(int i=2;i*i<=n;i++){
            if(primes[i]==1){
                for(int j=i*i;j<=n;j+=i){
                    primes[j] = 0;
                }
            }
        }
        return primes[n];
    }
    int countPrimeSetBits(int left, int right) {
        int cnt = 0;
        for(int i=left;i<=right;i++){
            int val = __builtin_popcount(i);
            if(isPrime(val)) cnt++;
        }
        return cnt;
    }

File: 762. Prime Number of Set Bits in Binary Representation.ts
function isPrime(n:number) {
    let primes:boolean[] = new Array(n + 1).fill(true);

    primes[0] = false;
    primes[1] = false;

    for (let i = 2; i * i <= n; i++) {
        if (primes[i] === true) {
            for (let j = i * i; j <= n; j += i) {
                primes[j] = false;
            }
        }
    }

    return primes[n];
}
function popcount(n:number) {
    let count:number = 0;
    while (n > 0) {
        count += (n & 1);
        n = n >> 1;
    }
    return count;
}
function countPrimeSetBits(left: number, right: number): number {
    let cnt:number = 0;

    for (let i:number = left; i <= right; i++) {
        let val:number = popcount(i);
        if (isPrime(val)) cnt++;
    }

    return cnt;
};